conjure/objectwriter.go

```diff
@@ -28,10 +29,33 @@ const (
 	dataVarName     = "data"
 )
 
+// logSafetyToAnnotation converts a LogSafety value to its corresponding safelogging annotation string
+func logSafetyToAnnotation(safety spec.LogSafety_Value) string {
+	switch safety {
+	case spec.LogSafety_SAFE:
+		return "@Safe"
+	case spec.LogSafety_UNSAFE:
+		return "@Unsafe"
+	case spec.LogSafety_DO_NOT_LOG:
+		return "@DoNotLog"
+	default:
+		panic("unhandled LogSafety value: " + safety)
+	}
+}
+
 func writeObjectType(file *jen.Group, objectDef *types.ObjectType) {
 	// Declare struct type with fields
 	containsCollection := false // If contains collection, we need JSON methods to initialize empty values.
-	file.Add(objectDef.Docs.CommentLine()).Type().Id(objectDef.Name).StructFunc(func(structDecl *jen.Group) {
+
+	// Compute the overall safety of the struct and add comment-based annotation
+	safetyCache := make(map[types.Type]spec.LogSafety)
```

Comment:
Let's move cache up one more level (add as parameter to this function) so that it can be shared across all `writeObjectType` calls